R/class-workbook.R

```diff
@@ -3468,6 +3468,43 @@ wbWorkbook <- R6::R6Class(
       )
     },
 
+    #' @description Add xml chart
+    #' @param sheet sheet
+    #' @param dims dims
+    #' @param xml xml
+    #' @returns The `wbWorkbook` object
+    add_chart_xml = function(
+      sheet = current_sheet(),
+      xml,
+      dims = "A1:H8"
+    ) {
+      sheet <- private$get_sheet_index(sheet)
+
+      dims_list <- strsplit(dims, ":")[[1]]
+      from <- col2int(dims_list)
+      to <- as.numeric(gsub("\\D+", "", dims_list))
+
+      next_chart <- NROW(self$charts) + 1
+
+      chart <- data.frame(
+        chart = xml,
+        colors = colors1_xml,
+        style = stylebarplot_xml,
+        rels = chart1_rels_xml(next_chart)
+      )
+
+      self$charts <- rbind(self$charts, chart)
+
+      # create Drawing
```

Comment:
for the initial implementation this is fine, though this needs to be improved to handle cases with existing drawings